src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/ThunkPool.cs

```diff
@@ -44,8 +45,10 @@ internal static class Constants
         public static readonly int ThunkCodeSize = RuntimeImports.RhpGetThunkSize();
         public static readonly int NumThunksPerBlock = RuntimeImports.RhpGetNumThunksPerBlock();
         public static readonly int NumThunkBlocksPerMapping = RuntimeImports.RhpGetNumThunkBlocksPerMapping();
-        public static readonly uint ThunkBlockSize = (uint)RuntimeImports.RhpGetThunkBlockSize();
-        public static readonly nuint ThunkBlockSizeMask = ThunkBlockSize - 1;
+        public static readonly uint ThunkCodeBlockSize = BitOperations.RoundUpToPowerOf2((uint)(ThunkCodeSize * NumThunksPerBlock));
+        public static readonly nuint ThunkCodeBlockSizeMask = ThunkCodeBlockSize - 1;
+        public static readonly uint ThunkDataBlockSize = BitOperations.RoundUpToPowerOf2((uint)(ThunkDataSize * NumThunksPerBlock));
```

Comment:
```suggestion
        public static readonly uint ThunkDataBlockSize = BitOperations.RoundUpToPowerOf2((uint)(ThunkDataSize * NumThunksPerBlock + IntPtr.Size));
```
Add `IntPtr.Size` to account for the next link so that this is correct by construction?

Reply:
Should not be necessary, you will still round up to the power of 2 ending on the page size, for example on windows, data size is 16 bytes and there is 250 thunks per block, so 4000 bytes and that will be rounded to page size, 4096 that is used by the arch implementation.

Reply:
I agree that it is not necessary for the current sizes, but it is not obvious how the extra pointer is accounted for.

I think this should have a comment about it and/or `+ IntPtr.Size`.